hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/AzureBlobFileSystemStore.java

```diff
@@ -842,7 +860,9 @@ public void createDirectory(final Path path, final FsPermission permission,
           !isNamespaceEnabled || abfsConfiguration.isEnabledMkdirOverwrite();
       Permissions permissions = new Permissions(isNamespaceEnabled,
           permission, umask);
-      final AbfsRestOperation op = client.createPath(getRelativePath(path),
+      createClient.createMarkerBlobs(path, overwrite, permissions, false, null,
```

Comment:
Nit: This is only needed in case of Blob Client right??
May be we can move this inside blobClient.createPath() function only and define a private method inside AbfsBlobCleint.

That w wa wwe won't need this in Abstract Parent and DFSClient with no implementation.

Design wise also it will look better.

Reply:
Since we need isNamespace info also, kept it here but corrected the conditions

Reply:
Updated code in AbfsBlobClient class